qiskit_nature/problems/sampling/protein_folding/peptide/chains/main_chain.py

```diff
@@ -0,0 +1,189 @@
+        main_chain_len: int,
+        main_chain_residue_sequences: List[str],
+        side_chain_lens: List[int],
+        side_chain_residue_sequences: List[str],
+    ) -> Sequence[MainBead]:
+        """
+        Creates a main chain for a given main chain length and side chain data.
+
+        Args:
+            main_chain_len: length of the main chain of a peptide.
+            main_chain_residue_sequences: list of characters that define residues for a main chain.
+            side_chain_lens: list of lengths of all side chains.
+            side_chain_residue_sequences: list of characters that define residues for all side
+            beads.
+
+        Returns:
+            An instance of a MainChain class.
+
+        Raises:
+            InvalidSizeException: when the length of list of side chain lengths provided does not
+                                    equal the length of the main chain.
+            InvalidSideChainException: when first, second or last main beads have a side chain.
+        """
+        main_chain = []
+        self._validate_chain_lengths(main_chain_len, side_chain_lens)
+        self._validate_side_chain_index_by_lengths(side_chain_lens)
+        self._validate_side_chain_index_by_residues(side_chain_residue_sequences)
+
+        for main_bead_id in range(main_chain_len - 1):
+            bead_turn_qubit_1 = self._build_turn_qubit(main_chain_len, 2 * main_bead_id)
+            bead_turn_qubit_2 = self._build_turn_qubit(main_chain_len, 2 * main_bead_id + 1)
+            side_chain = self._create_side_chain(
+                main_bead_id, main_chain_len, side_chain_lens, side_chain_residue_sequences
+            )
+            main_bead = MainBead(
+                main_bead_id,
+                main_chain_residue_sequences[main_bead_id],
+                [bead_turn_qubit_1, bead_turn_qubit_2],
+                side_chain,
+            )
+            main_chain.append(main_bead)
+        main_bead = MainBead(main_chain_len - 1, None, None, None)
+        main_chain.append(main_bead)
+        return main_chain
+
+    @staticmethod
+    def _validate_chain_lengths(main_chain_len: int, side_chain_lens: List[int]) -> None:
+        if side_chain_lens is not None and main_chain_len != len(side_chain_lens):
+            raise InvalidSizeException(
+                f"The length of list of side chain lengths provided: {len(side_chain_lens)}, "
+                f"does not equal the length of the main chain: {main_chain_len}"
+            )
+
+    @staticmethod
+    def _validate_side_chain_index_by_lengths(side_chain_lens: List[int]) -> None:
+        if side_chain_lens is not None and (
+            side_chain_lens[0] != 0 or side_chain_lens[1] != 0 or side_chain_lens[-1] != 0
+        ):
+            raise InvalidSideChainException(
+                "First, second and last main beads are not allowed to have a side chain. Non-zero "
+                "length provided for an invalid side chain."
+            )
```

Comment:
In the docstring of the `protein_folding` module you are saying that only side-chains of length 1 are supported. Shouldn't this be tested here?

Or in other words: what happens when a user tries to specify longer side chains? Will the code exit gracefully with a warning/error message or will it run but produce non-sensical results? (or will it even simply crash with some traceback?)